includes/class-wcpdf-settings-callbacks.php

```diff
@@ -412,20 +428,31 @@ public function media_upload( $args ) {
 	 */
 	public function next_number_edit( $args ) {
 		extract( $args ); // $store, $size, $description
-		// Sequential_Number_Store object
-		if( is_object( $store ) ) {
-			$next_number         = $store->get_next();
-			$store               = $store->store_name;
-		// legacy
-		} else {
-			$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method(); 
-			$number_store        = new Sequential_Number_Store( $store, $number_store_method ); 
-			$next_number         = $number_store->get_next();
+
+		$store_name  = '';
+		$next_number = '';
+
+		if( ! empty( $store_callback ) ) {
+			// Sequential_Number_Store object
+			if ( $store_callback instanceof Sequential_Number_Store ) {
+				$store = $store_callback;
+			// array with document object + function name
+			} elseif ( is_array( $store_callback ) ) {
+				$store = ! empty( $store_callback_args ) ? call_user_func_array( $store_callback, $store_callback_args ) : call_user_func( $store_callback );
+			// legacy
+			} else {
+				$number_store_method = WPO_WCPDF()->settings->get_sequential_number_store_method(); 
+				$store               = new Sequential_Number_Store( $store_callback, $number_store_method ); 
+			}
```

Comment:
this will break existing extensions using the `store` parameter. Handling this should be done the same as the other `*_callback` parameters. So:
```php
if( ! empty( $store_callback ) ) {
	$store = ! empty( $store_callback_args ) ? call_user_func_array( $store_callback, $store_callback_args ) : call_user_func( $store_callback );
}
```

That way all the rest of the code can stay untouched.

Reply:
maybe `isset()` is better for `$store_callback_args` (same for the other ones)

Reply:
Please see my commit https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/254/commits/030fa5af1a9c42771107232ced15487ae5ed5a08, I follow this path because of this comment: https://wpovernight.slack.com/archives/DV2FX6YU8/p1640249171003800